jupyterhub/apihandlers/base.py

```diff
@@ -62,6 +85,43 @@ def check_referer(self):
             return False
         return True
 
+    def get_scope_filter(self, req_scope):
+        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return.
+        Filter is a callable that takes a resource name and outputs true or false"""
+        kind_regex = re.compile(r':?(user|service|group)s:?')
+        try:
+            kind = re.search(kind_regex, req_scope).group(1)
+        except AttributeError:
+            self.log.warning(
+                "Regex error while processing scope %s, throwing 500", req_scope
+            )
+            raise web.HTTPError(
+                log_message="Unrecognized scope guard on method: %s" % req_scope
+            )
+        try:
+            sub_scope = self.parsed_scopes[req_scope]
+        except AttributeError:
+            raise web.HTTPError(
+                403,
+                "Resource scope %s (that was just accessed) not found in parsed scope model"
+                % req_scope,
+            )
+
+        def has_access(resource_name):
+            if sub_scope == scopes.Scope.ALL:
+                found_resource = True
+            else:
+                found_resource = resource_name in sub_scope[kind]
+            if not found_resource:  # Try group-based access
+                if 'groups' in sub_scope and kind == 'users':
+                    user = self.current_user()
```

Comment:
I think this lookup is incorrect (current_user is the requesting user, not the one we are checking against), but passing `resource` to has_access should fix this.